x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1105,15 +1169,39 @@ public void findActiveTokensForRealm(String realmName, @Nullable Predicate<Map<S
      */
     public void findActiveTokensForUser(String username, ActionListener<Collection<Tuple<UserToken, String>>> listener) {
         ensureEnabled();
-        final SecurityIndexManager frozenSecurityIndex = securityIndex.freeze();
         if (Strings.isNullOrEmpty(username)) {
             listener.onFailure(new IllegalArgumentException("username is required"));
-        } else if (frozenSecurityIndex.indexExists() == false) {
+            return;
+        }
+        final Instant now = clock.instant();
+        findActiveTokensForUser(username, securityTokensIndex, now, ActionListener.wrap(tokens -> {
+            if (false == securityTokensIndex.indexExists()
+                    || Instant.now().minus(ExpiredTokenRemover.MAXIMUM_TOKEN_LIFETIME_HOURS, ChronoUnit.HOURS)
+                            .isBefore(securityTokensIndex.getCreationTime())) {
+                findActiveTokensForUser(username, securityMainIndex, now, ActionListener.wrap(tokensInSecurityMain -> {
```

Comment:
As above - is it better to do this as 1 search?

Reply:
I will try to explore this alternative.

Reply:
I have pushed https://github.com/elastic/elasticsearch/pull/40742/commits/4e135c34894544b8753ce0968f154f3c8a0e9931 . I'm not terribly happy about it, but it's the best I could come up with.